Channel/messaging on Android, Pods paths, FriendlyName

jukben opened the pull request:
Thanks for the work on this, here are my changes what I need to apply to get this library closer to the production-ready into my project. It might be helpful for some of you.

- added FriendlyName into the SESSION_STARTED event
- added Pods into Framework's paths
- Implemented (basic) channel/messaging on Android platform
- fixed `tint` on Android platform

Commit:
First commit

Commit:
Create README.md

Commit:
chromecast - android support!

Commit:
Merge branch 'develop' of https://github.com/charliesbox/react-native-google-cast into develop

Commit:
starting a good docs

Commit:
renaming android implementation

Commit:
renaming android implementation

Commit:
preparing library

Commit:
preparing library

Commit:
preparing library

Commit:
preparing library

Commit:
typo :)

Commit:
temporal fix while checking constants on android

Commit:
refactoring

Commit:
refactoring

Commit:
refactoring

Commit:
refactoring

Commit:
typo with name lol

Commit:
renaming

Commit:
removing dependencies

Commit:
removing dependencies

Commit:
removing description

Commit:
checking if chromecast is connected and initialized

Commit:
checking if chromecast is connected and initialized

Commit:
sending new events

Commit:
almost done!

Commit:
chromecast for iOS - READY

Commit:
chromecast for iOS - READY

Commit:
chromecast for iOS - READY

Commit:
Update README.md

Commit:
Update README.md

Commit:
cleaning up

Commit:
changing milliseconds to seconds as a new standard

Commit:
cleanning up

Commit:
getting stream position in iOS

Commit:
getting stream position in android

Commit:
renaming functions - startscan

Commit:
readme

Commit:
Update README.md

Commit:
readme

Commit:
Update README.md

Commit:
attempt to use cocoapods

Commit:
attempt to use cocoapods

Commit:
attempt to use cocoapods

Commit:
attempt to use cocoapods

Commit:
attempt to use cocoapods

Commit:
attempt to use cocoapods

Commit:
attempt to use cocoapods

Commit:
attempt to use cocoapods

Commit:
attempt to use cocoapods

Commit:
attempt to use cocoapods

Commit:
attempt to use cocoapods

Commit:
attempt to use cocoapods

Commit:
Update README.md

Commit:
Update README.md

Commit:
0.0.2

Commit:
fixing typos

Commit:
0.0.3

Commit:
adding events and constants

Commit:
0.0.4

Commit:
cocoapods

Commit:
cocoapods

Commit:
Update README.md

Commit:
Update README.md

Commit:
cocoapods

Commit:
Merge branch 'develop'

Commit:
Merge branch 'master' of https://github.com/charliesbox/react-native-google-cast

Commit:
adding constants to readme

Commit:
0.1.0

Commit:
adding constants to android

Commit:
0.1.1

Commit:
Update README.md

Commit:
adding constants to ios

Commit:
Merge branch 'master' of https://github.com/charliesbox/react-native-google-cast

Commit:
0.1.2

Commit:
disconnect callback fix

Commit:
0.1.3

Commit:
Update README.md

Commit:
fixed issues while disconnecting

Commit:
Merge branch 'master' of https://github.com/charliesbox/react-native-google-cast

Commit:
fixed issues while disconnecting

Commit:
fixed issues while disconnecting

Commit:
Update README.md

Commit:
0.2.0

Commit:
Merge branch 'master' of https://github.com/charliesbox/react-native-google-cast

Commit:
Update README.md

Commit:
0.2.1

Commit:
using react-native-library as a template

Commit:
added example inside the project

Commit:
added example inside the project

Commit:
fixed glitch when getting dependencies from cocoapods

Commit:
added example inside the project

Commit:
Update README.md

Commit:
added example inside the project

Commit:
Merge branch 'feature/rnpm' of https://github.com/charliesbox/react-native-google-cast into feature/rnpm

Commit:
New react-native link instructions!

Commit:
1.0.0

Commit:
device disconnected feature

Commit:
device disconnected support for android and iOS

codetony25 committed:
moved array declaration inside loop to avoid same referencing to fix duplicated scanned devices in ios

Commit:
Merge pull request #17 from codetony25/master

moved array declaration inside loop to avoid same referencing to fix …

Commit:
Merge branch 'master' into feature/new-functions

# Conflicts:
#	ios/GoogleCast.m

Commit:
fixes to dependencies warnings

Commit:
1.2.0

petrbela committed:
compatibility with RN 0.40+

Commit:
Merge pull request #24 from petrbela/rn-0.40

compatibility with RN 0.40+

Commit:
Fix bug when two chromecast devices are connected

Commit:
Merge pull request #27 from llewan/bug/manyDevices

Fix bug when two chromecast devices are connected

Commit:
upgraded example and module version

Commit:
Update README.md

Commit:
Update README.md

Commit:
added extra dependencies into example project

Commit:
Update README.md

Commit:
added extra dependencies into example project

Commit:
added extra dependencies into example project

vmseba committed:
Added support for specific app id.

vmseba committed:
unified start scan functions

vmseba committed:
fixed startScan repeated key

vmseba committed:
Fixed implementation of startScan.

vmseba committed:
Fixed import in android project.

Commit:
Merge pull request #29 from vmseba/master

Support for specific app id.

DracoBlue committed:
Changed podfile example to version "2.10.4.1"

Since it will crash with 3.3.x of google cast sdk on iOS

Commit:
Merge pull request #38 from DracoBlue/patch-1

Changed podfile example to version "2.10.4.1"

Commit:
upgraded sdk for iOS to v3

petrbela committed:
basic support for v3 on iOS

petrbela committed:
iOS: default expanded controller and introductory overlay, session events

petrbela committed:
update constants

petrbela committed:
Android basic prototype

petrbela committed:
Android cast button working, upgrade to latest React Native

petrbela committed:
added playback functions on iOS

petrbela committed:
media events & custom channel support

petrbela committed:
reorganize example project dependencies and files

This simplifies installation for developing the library and avoids the need for symlinking, which causes issues in RN.
Also provides an explicit podspec for CocoaPods installation, which resolves issue with the library not seeing the google-cast-sdk pod during compilation.

petrbela committed:
implement `getCastState` and `endSession` for iOS and Android

petrbela committed:
Implement MEDIA_PLAYBACK_STARTED/ENDED events correctly

petrbela committed:
call playback methods on main thread

petrbela committed:
less strict dependency version

petrbela committed:
update to build version 25 with instructions on conflict resolution

petrbela committed:
Enable setting tintColor of the CastButton

jukben committed:
fix(android): fix Cast button color

jukben committed:
feat: added friendlyName into SESSION_STARTED event

jukben committed:
feat(android): implemented messages

jukben committed:
fix(ios): added Pods into framework paths

petrbela commented:
Hi Jakub, thanks for the contribution! I'll be working on getting this lib to production-level later this week and I'll make sure to include your PR.

jukben commented:
Awesome! Let me know if there is something I can help with. 

Just be aware that I haven't implemented (to match iOS 100%) on Android side:
- GoogleCast.CHANNEL_CONNECTED,
- GoogleCast.CHANNEL_DISCONNECTED

jukben committed:
fix(android): unify API with iOS

iOS is using `playPosition`, but Android had implemented `seconds`. This unified the experience.
